optimum/intel/ipex/modeling_base.py

```diff
@@ -561,6 +590,18 @@ def forward(
 
         return CausalLMOutputWithPast(logits=logits, past_key_values=past_key_values)
 
+    def _prepare_generation_config(
+        self, generation_config: Optional[GenerationConfig], **kwargs: Dict
+    ) -> Tuple[GenerationConfig, Dict]:
+        generation_config, model_kwargs = self.model_cls._prepare_generation_config(self, generation_config, **kwargs)
+        generation_method = generation_config.get_generation_mode(kwargs.get("assistant_model", None)).value
```

Comment:
assistant_model will not be part of the kwargs when calling `generate` https://github.com/huggingface/transformers/blob/bd5091df8db7cea1a9f94f797fc11487f840ade1/src/transformers/generation/utils.py#L1628 so I think we can remove
 ```suggestion
        generation_method = generation_config.get_generation_mode().value
```

Reply:
also assisted decoding is not supported for patched model is this correct? if yes we can also add a warning when given when calling `generate`

Reply:
But the assistant model is required when determine generation mode, see [here](https://github.com/huggingface/transformers/blob/bd5091df8db7cea1a9f94f797fc11487f840ade1/src/transformers/generation/utils.py#L1777-L1778)